daemon/daemon.go

```diff
@@ -1604,8 +1604,21 @@ func (d *Daemon) GetNodeSuffix() string {
 	return ip.String()
 }
 
-// ClearPolicyConsumers removes references to the specified id from the rules in
-// the daemon's policy repository.
-func (d *Daemon) ClearPolicyConsumers(id uint16) *sync.WaitGroup {
-	return d.policy.RemoveEndpointIDFromRuleCaches(id)
+// ReleaseIdentity asynchronously removes references to the specified identity
+// from the rules in the daemon's policy repository.
+func (d *Daemon) ReleaseIdentity(identity *identity.Identity) {
```

Comment:
Nit: this function naming collides with that of the allocator - I think this should be changed to something like `RemoveIdentityFromPolicyCaches` - this is verbose, but more clear.

Reply:
This should be resolved when combining with #7539 .

Reply:
FYI these bits were stripped out when I rebased against identitymanager (#7616).